### PR TITLE
RedCanary - improve endpoint context standard handling

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/25624

## Description
The issue was RedCanary returns `None` under `mac_address` sometimes, and it was not handled appropriately. 

I've refactored the whole function as it is not protective at all

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
See failing test at https://app.circleci.com/pipelines/github/demisto/content/16362/workflows/e8e21046-45c9-4396-933f-134029229c81/jobs/58534/parallel-runs/1?filterBy=FAILED

### CODE REVIEW

File: Packs/RedCanary/Integrations/RedCanary/RedCanary.py [hunk 1]
```diff
@@ -111,21 +111,39 @@ def playbook_name_to_id(name):
...
+            'OS': endpoint_attributes.get('platform'),
+            'OSVersion': endpoint_attributes.get('operating_system'),
+            'IsIsolated': endpoint_attributes.get('is_isolated'),
+            'IsDecommissioned': endpoint_attributes.get('is_decommissioned'),
+        }
+        ip_addresses = []
+        mac_addresses = []
+        for address in endpoint_attributes.get('endpoint_network_addresses', []):
+            address_attributes = address.get('attributes', {})
+            ip_address_object = address_attributes.get('ip_address', {})
```
Comment (ShahafBenYakir) [inline Packs/RedCanary/Integrations/RedCanary/RedCanary.py:131]:
```suggestion
            if address_attributes:
                ip_address_object = address_attributes.get('ip_address', {})
```

File: Packs/RedCanary/Integrations/RedCanary/RedCanary.py [hunk 2]
```diff
@@ -111,21 +111,40 @@ def playbook_name_to_id(name):
...
+        ip_addresses = []
+        mac_addresses = []
+        for address in endpoint_attributes.get('endpoint_network_addresses', []):
+            address_attributes = address.get('attributes', {})
+            if address_attributes:
+                ip_address_object = address_attributes.get('ip_address', {})
+                if ip_address_object:
+                    ip_address_attributes = ip_address_object.get('attributes', {})
+                    if ip_address_attributes:
+                        ip_addresses.append(ip_address_attributes.get('ip_address'))
```
Comment (ShahafBenYakir) [inline Packs/RedCanary/Integrations/RedCanary/RedCanary.py:136]: Actually why nor using `demisto.get()`?

```suggestion
              ip_address = demisto.get(address_attributes, 'ip_address.attributes.ip_address')
              if  ip_address:
                   ip_addresses.append(ip_address)
```

Comment (Itay4) [reply Packs/RedCanary/Integrations/RedCanary/RedCanary.py:136]: not big fan of it

**Comment by kirbles19** [comment]
@Itay4 Done.

**Pull request merged**